src/index.ts

```diff
@@ -6,10 +6,34 @@ import {
   DecorationSet,
   Decoration,
 } from '@codemirror/view';
+import type { StyleSpec } from 'style-mod';
+import { Extension } from '@codemirror/state';
 import { syntaxTree } from '@codemirror/language';
 import { Range } from '@codemirror/rangeset';
 import { namedColors } from './named-colors';
 
+interface CSSColorPickerOptions {
+  /**
+   * Additional [`style-mod`](https://github.com/marijnh/style-mod#documentation)
+   * style spec providing theme for the color picker.
+   */
+  style?: {
+    /** Style spec for the color picker `<div>` container */
+    wrapper?: StyleSpec;
+    /** Style spec for the color picker `<input>` element */
+    input?: StyleSpec;
+  };
+}
+
+interface PickerState {
+  from: number;
+  to: number;
+  alpha: string;
+  colorType: ColorType;
+}
+
+const pickerState = new WeakMap<HTMLInputElement, PickerState>();
```

Comment:
This is probably cleaner, but just wondering, any other reason to use this vs `dataset`. Is it so that it's typed?

Reply:
no typecasting between string, empty string, and numbers